src/hp/magicsock/conn.rs

```diff
@@ -478,6 +478,23 @@ impl Conn {
     }
 }
 
+impl Drop for Conn {
+    fn drop(&mut self) {
+        if self.is_closed() {
+            return;
+        }
+        // the actor holds a strong ref to us, so 2 means we're the last conn
+        if Arc::strong_count(&self.inner) > 2 {
+            return;
+        }
+        self.inner.closing.store(true, Ordering::Relaxed);
+        self.inner.closed.store(true, Ordering::SeqCst);
```

Comment:
This leaves things in a less than ideal place, as the assumption is that things have been cleanly stopped once `closed = true`. So we should improve the internal structure if we are doing shutdown like this.

Reply:
I just copied the code. But who will ever read these flags after this? The actor task is aborting, and the last Conn has just been dropped.

It would only matter if the actor task is in some sync code (which can not be aborted) and then reads these flags.